liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/StartDebugMojoSupport.java

```diff
@@ -218,13 +346,45 @@ private void writeBootstrapProperties(File file, Map<String, String> properties)
         }
     }
 
-    private void writeJvmOptions(File file, List<String> options) throws IOException {
+    private void writeServerEnvProperties(File file, Map<String, String> mavenProperties) throws IOException {
         makeParentDirectory(file);
         PrintWriter writer = null;
         try {
             writer = new PrintWriter(file, "UTF-8");
             writer.println(HEADER);
-            for (String option : options) {
+            for (Map.Entry<String, String> entry : mavenProperties.entrySet()) {
+                String key = entry.getKey();
+                writer.print(key);
+                writer.print("=");
+                String value = entry.getValue();
+                writer.println((value != null) ? value.replace("\\", "/") : "");
+                if (value == null) {
+                    log.error("The value of the server.env property " + entry.getKey() + " is null. Verify if the needed POM properties are set correctly.");
```

Comment:
If this is not intended to fail the build, it should be logged as a warning rather than an error. But in what circumstances would be expect to get a null value from the entry set?

Reply:
That was existing behavior in the other write methods, but I see your point. I will change them all to warnings.